src/types.c

```diff
@@ -50,6 +50,7 @@ bool initText(Text* self, const char* str, SDL_Color color) {
       printf("Unable to create texture from rendered text! SDL Error: %s\n",
              SDL_GetError());
     } else {
+      if (self->origin) SDL_DestroyTexture(self->origin);
```

Comment:
Any specific reason for adding this line and the line in `createText`? Will lacking of these lines cause a memory leak? A restriction I want to maintain is that `initX` should be called on uninitialized memory and `createX` should malloc an area of uninitialized memory and then initialize it.

Reply:
there is a function named `setText` at types.c, it called `initText` every time the text changed.
In this case, the texture is repeatedly created when `initText` is called, resulting in a memory leak.

Reply:
You're right. Then can you move the line to `setText` thus leaving `createText` and `initText` unchanged? `initText` shouldn't assume the memory area has any valid address. The `if` statement breaks that.

Reply:
I don't think that's a good idea. It can avoid possible memory leaks in the future. But if you insist, I can change it.

Reply:
Start a new PR.